packages/firestore/src/remote/remote_store.ts

```diff
@@ -558,12 +558,23 @@ export class RemoteStore implements TargetMetadataProvider {
     // Record the stream token.
     return this.localStore
       .setLastStreamToken(this.writeStream.lastStreamToken)
-      .then(() => {
-        // Send the write pipeline now that the stream is established.
-        for (const batch of this.writePipeline) {
-          this.writeStream.writeMutations(batch.mutations);
+      .then(
+        () => {
+          // Send the write pipeline now that the stream is established.
+          for (const batch of this.writePipeline) {
+            this.writeStream.writeMutations(batch.mutations);
+          }
+        },
+        err => {
+          if (err.code === Code.FAILED_PRECONDITION) {
```

Comment:
FYI: Instead of relying on the error code, I tried to make this more specific and added a new error class at one point. I wanted to do an `instanceof CustomPrimaryLeaseLostError` check here. It turns out that instanceof checks don't work with errors though, and so I abandoned this approach.